### ISSUE TITLE
Create helper methods for the /contactdb endpoint

### ISSUE THREAD

**Issue by thinkingserious** [sendgrid/sendgrid-ruby#311]
#### Issue Summary

Please see `USE_CASES.md` for current example of existing helper methods.

You may find simple example code for all of the `/contactdb` endpoint functionality in the `USAGE.md` file under `CONTACTDB` in the Table of Contents. Complete interactive documentation can be found [here](https://sendgrid.com/docs/API_Reference/api_v3.html).

Before writing any code, we suggest you review our `CONTRIBUTING.md` file and propose your solution for discussion here in the comments. Thank you!

#### Aceptance Criteria

* helper methods are created for all documented /contactdb functionality
* tests pass locally, in Travis CI and using Docker
* updated `USE_CASES.md` or `/use-cases` documentation

**Comment by DearGrenadier** [comment]
mind if I take it?

**Comment by DearGrenadier** [comment]
@thinkingserious mind if I take?

**Comment by thinkingserious** [comment]
Thank you @DearGrenadier, it's all yours!

**Comment by DearGrenadier** [comment]
Hey @thinkingserious , I've investigated all proposed by you resources and came up with some idea how to achieve the goal we need. Firstly I wanna clarify and highlight all endpoints, which might be used along with helpers. I think that's following guys(selected):

- [x] POST /contactdb/custom_fields
- [ ] GET /contactdb/custom_fields
- [x] GET /contactdb/custom_fields/{custom_field_id}
- [x] DELETE /contactdb/custom_fields/{custom_field_id}
- [x] POST /contactdb/lists
- [ ] GET /contactdb/lists
- [x] DELETE /contactdb/lists
- [x] PATCH /contactdb/lists/{list_id}
- [x] GET /contactdb/lists/{list_id}
- [x] DELETE /contactdb/lists/{list_id}
- [x] POST /contactdb/lists/{list_id}/recipients
- [x] GET /contactdb/lists/{list_id}/recipients
- [x] POST /contactdb/lists/{list_id}/recipients/{recipient_id}
- [x] DELETE /contactdb/lists/{list_id}/recipients/{recipient_id}
- [x] PATCH /contactdb/recipients
- [x] POST /contactdb/recipients
- [x] GET /contactdb/recipients
- [x] DELETE /contactdb/recipients
- [ ] GET /contactdb/recipients/billable_count
- [ ] GET /contactdb/recipients/count
- [x] GET /contactdb/recipients/search
- [x] GET /contactdb/recipients/{recipient_id}
- [x] DELETE /contactdb/recipients/{recipient_id} +
- [x] GET /contactdb/recipients/{recipient_id}/lists
- [x] GET /contactdb/reserved_fields
- [x] POST /contactdb/segments
- [ ] GET /contactdb/segments
- [x] PATCH /contactdb/segments/{segment_id}
- [x] GET /contactdb/segments/{segment_id}
- [x] DELETE /contactdb/segments/{segment_id}
- [x] GET /contactdb/segments/{segment_id}/recipients

Actually, not selected endpoints don't receive any args and I think their usage will be the same as it is. Regarding the implementation I think its gonna be smth pretty similar to `mail` helpers, will be some primitive classes, which just hold information, and some generic structures able to group primitives. I'd suggest to start implement needed helpers for one of endpoint groups above, for instance `custom_fields`, and then, in progress, decide exactly what we need, based on some start results. I'm gonna create WIP PR and post there some intermediate development progress. Please let me know what do you think regarding all of this.

**Comment by thinkingserious** [comment]
That's awesome, thanks @DearGrenadier!

**Comment by DearGrenadier** [comment]
I've unchecked all endpoints list items, just to monitor progress.

**Comment by DearGrenadier** [comment]
@thinkingserious Just wanna clarify couple of concerns:
1. I was pocking a bit at segments API stuff, current implementation of client, I mean how it's used for contactdb requests, and I'm not sure that it make sense to add more helpers to this part of API(considering only segments atm), because rest of endpoints are pretty straight forward.
2. Also I checked sources of your http client and found there internal [casting of data to json](https://github.com/sendgrid/ruby-http-client/blob/master/lib/ruby_http_client.rb#L152), didn't include it to my changes, is that fair?
3. I chose USAGE.md for example updates, that's quite suitable place I think, isnt it? If I'd chose USE_CASES.md I had to copy some content from USAGE.md not sure if that worth to do.
